codegen/validation.go

```diff
@@ -41,67 +83,23 @@ func init() {
 	userValT = template.Must(template.New("user").Funcs(fm).Parse(userValTmpl))
 }
 
-// HasValidations returns true if the given attribute or any of its children
-// recursively has validations. If ignoreRequired is true then HasValidation
-// does not consider "Required" validations. This is necessary to know whether
-// validation code should be generated for types that don't use pointers to
-// define required fields.
-func HasValidations(att *expr.AttributeExpr, ignoreRequired bool) bool {
-	if att.Validation != nil {
-		if !ignoreRequired || !att.Validation.HasRequiredOnly() {
-			return true
-		}
-	}
-	if o := expr.AsObject(att.Type); o != nil {
-		for _, catt := range *o {
-			seen := make(map[*expr.AttributeExpr]struct{})
-			seen[att] = struct{}{}
-			if hasValidationsRecurse(catt.Attribute, ignoreRequired, seen) {
-				return true
-			}
-		}
-	}
-	return false
-}
-
 // ValidationCode produces Go code that runs the validations defined in the
 // given attribute definition if any against the content of the variable named
 // target. The generated code assumes that there is a pre-existing "err"
 // variable of type error. It initializes that variable in case a validation
 // fails.
 //
```

Comment:
Please describe the role of the `ah` argument (which should now be `an` I guess).